src/installation/installing-rust-for-esp.md

```diff
@@ -0,0 +1,135 @@
+- MSVC: This is the recommended ABI. When installing `rustup`, it will check if all the requirements are installed, and, if they are not, it allows the user to install them.
+- GNU: No checks are done in `rustup` and expect that the user takes care of properly installing it.
+
+[rustup]: https://rustup.rs/
+[running Windows as your host operating system, you must also install one of the available ABIs]: https://rust-lang.github.io/rustup/installation/windows.html
+
+## ldproxy
+
+`ldproxy` crate is required when building applications using the Rust standard library, `std`. To install:
+```sh
+cargo install ldproxy
+```
+
+This tool is required regardless of the target architecture.
+
+[`ldproxy` crate]: https://github.com/esp-rs/embuild/tree/master/ldproxy
+
+## RISC-V
+
+If you only want to target `RISC-V` chips, installation is simpler. In order to build
+applications for `RISC-V` targets we need to use a [Rust nightly toolchain] with the `rust-src` [component], both things can be installed with:
+
+```bash
+rustup toolchain install nightly --component rust-src
+```
+
+For bare-metal, target can be installed by running:
+
+```bash
+rustup target add riscv32imc-unknown-none-elf
+```
+
+For `std` applications, the `riscv32imc-esp-espidf` target does not have prebuilt objects distributed through rustup, therefore the `-Z build-std` [unstable cargo feature] is required within your project. This [unstable cargo feature] can also be added to `.cargo/config.toml` of your project. Our [template projects], which we will later discuss, already take care of this.
+
+Also, when building `std` applications, make sure you have [`LLVM`] and [`ldproxy`] installed.
+
+At this point, you are ready to build applications for all the Espressif chips based on RISC-V architecture.
+
+The installation of Rust for ESP `RISC-V` targets can also be handled by `espup`, a tool that will be introduced
+in the [`espup` section].
+
+[Rust nightly toolchain]: https://rust-lang.github.io/rustup/concepts/channels.html#working-with-nightly-rust
+[component]: https://rust-lang.github.io/rustup/concepts/components.html
+[template projects]: ../writing-your-own-application/generate-project-from-template.md
+[unstable cargo feature]: https://doc.rust-lang.org/cargo/reference/unstable.html
+[`LLVM`]: https://llvm.org/
+[`ldproxy`]: #ldproxy
+[`espup` section]: #espup
+
+
+## Xtensa
+
+Because there is no `Xtensa` support in the mainline Rust compiler you must use the [esp-rs/rust] fork instead. There are a few options available for installing this compiler fork.
+
+- The recommended one is using `espup`. See [`espup` section] for more details.
+- Using [esp-rs/rust-build] installation scripts. This was the recommended way in the past, but now, the installation scripts are feature frozen and all the new features will only be included in `espup`. See the repository readme for instructions.
```

Comment:
Might be worth mentioning that, at least for the time being, `esp-rs/rust-build` will be maintained, however no new features will be added.